### ISSUE TITLE
copy of ModelContainer does not transfer much of the meta info

### ISSUE THREAD

**Issue by stscieisenhamer** [STScI-JWST/jwst#1844]
# Issue
One aspect of `ModelContainer` when it is created from an association is that a number of `meta` values are populated. However, if a `DataModel.copy` is performed, all that info is lost. Example:
```
>>> asn_container = jwst.datamodels.open('single_spec3_asn_new.json')
>>> asn_container.meta.instance
{'asn_rule': 'DMS_Level3_Base',
 'asn_table': {'asn_id': 'a3001',
  'asn_pool': 'none',
  'asn_rule': 'DMS_Level3_Base',
  'asn_type': 'None',
  'code_version': '0.9.12',
  'constraints': 'No constraints',
  'degraded_status': 'No known degraded exposures in association.',
  'products': [{'members': [{'expname': 'level2b_new/det_image_seq3_MIRIFULONG_34MEDIUMexp1_cal.fits',
      'exptype': 'science'}],
    'name': 'full'}],
  'program': 'noprogram',
  'target': 'none',
  'version_id': None},
 'asn_type': 'None',
 'date': '2018-03-29T11:17:25.725',
 'model_type': 'ModelContainer',
 'pool_name': 'none',
 'program': 'noprogram',
 'resample': {'output': 'full'},
 'table_name': '/Users/eisenham/Documents/ssbdev/testdata/pandokia/jwcalibdev/data4/jwst_test_data/miri/test_datasets/mrs/simulated/single_spec3_asn_new.json',
 'targname': 'none'}
>>> new_asn_container = asn_container.copy()
>>> new_asn_container.meta.instance
{'date': '2018-03-29T11:17:43.139', 'model_type': 'ModelContainer'}
```
# Resources
- Related: #1825

**Comment by jdavies-st** [comment]
Ack!

Why are we copying `ModelContainers`?  Are we expecting a deep copy or shallow copy?

The problem is that we want to store the asn info in a `dict`, but we don't know what the schema for that information is, as associations can contain a variety of things.  So since there is no schema defined for the assocation `meta`, then it does not get copied.

It also looks like none of the items in the actual container schema get copied either.  That seems like a bug.

**Comment by stscieisenhamer** [comment]
The copy is happening because of the general-usage-though-not-designed of `Step`'s copying their input. Though, the only reason this particular bug came through because `OutlierDetectionStep`, if given only one exposure, does not do detection, but does the copy. Otherwise, if detection does occur, the input is simply returned. Looking more closely, this, in itself, could be considered another bug.

**Comment by jdavies-st** [comment]
Yes, maybe the input should simply be returned in `OutlierDection` instead of copied.

Perhaps deep copy should not work on model containers?  As originally designed, it did, and `copy()` did go through and do a thorough deep copy.  This behavior was changed as part of #10127.  As you can see in the comments, that change was controversial.  =)

**Comment by jdavies-st** [comment]
This is definitely a bug in the `copy()` method.  I will fix.  We can discuss elsewhere whether it's a good idea to copy a `ModelContainer`.  =)

**Comment by hbushouse** [comment]
Regarding the lack of ASN schemas: perhaps it's time to define level-2b and level-3 ASN schemas, to contain all of those pertinent pieces of meta like asn_pool, asn_type, asn_rule, etc.

**Comment by hbushouse** [comment]
@jdavies-st It's fine to fix the copy() behavior so that it copies everything, but even after you've fixed that, folks should still think very long and hard before actually copying a ModelContainer, especially if it ever goes back to doing a deep copy.
